### PR TITLE
Enhancement/code coverage

### LINKED ISSUE #18: Write more tests
- meaningful tests, no dry-run “tests”
- at least 90% code coverage
- analytic reference tests on small inputs if possible

### PR DESCRIPTION
**Pull request by Enea09**

### CODE REVIEW

File: test/test_informationgain.jl [hunk 1]
```diff
@@ -34,12 +34,25 @@ end
...
+@testset "best_split" begin
     X = [1.0 2.0; 3.0 4.0; 2.5 0.5; 4.0 3.0]
     y = [0, 1, 0, 1]
     best_feature, best_threshold = DecisionTreeAndRandomForest.best_split(X, y)
     @test best_feature == 1
-    @test best_threshold == 2.5
+    @test best_threshold == 3.0
+end
+
+@testset "split_gini_categorical" begin
```
Comment (cedric-len) [inline test/test_informationgain.jl]: This should be called split_ig_categorical

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ on:
   push:
     branches:
       - main
+      - enhancement/code-coverage
```
Comment (cedric-len) [inline .github/workflows/CI.yml]: I guess this can be removed before merging right?

File: test/test_classificationtree.jl [hunk 3]
```diff
@@ -30,7 +30,69 @@
     print(tree)
```
Comment (cedric-len) [inline test/test_classificationtree.jl]: We should maybe remove all the print statements to increase performance

**Review by cedric-len — CHANGES_REQUESTED** [review]

File: test/test_informationgain.jl (hunk 1 above)
Comment (Enea09) [reply test/test_informationgain.jl]: Yes just noticed it as well. Was about to change it now :)

File: .github/workflows/CI.yml (hunk 2 above)
Comment (Enea09) [reply .github/workflows/CI.yml]: I think so

File: src/split-criterions/VarianceReduction.jl [hunk 4]
```diff
@@ -10,7 +10,7 @@ Calculate the sample variance of a given set of labels. It uses the standard for
 - `Float64`: The sample variance of the input label vector `y`.
 """
 function variance(y::AbstractVector)
-    variance = sum((y .- mean(y)) .^ 2) / length(y) - 1
+    variance = sum((y .- mean(y)) .^ 2) / (length(y) - 1)
```
Comment (marleenlukei) [inline src/split-criterions/VarianceReduction.jl:13]: That was indeed an error on my side. Good that you noticed it :)

**Review by marleenlukei — APPROVED** [review]
After the print statements have been removed, this can be merged. Maybe in our next call we can talk about whether we add some documentation on our evaluation metrics? (accuracy, r-squared score, mean squared error and their acceptable values)

**Pull request merged**